### ISSUE TITLE
FMS release 2022.04

### ISSUE THREAD

**Issue by DusanJovic-NOAA** [JCSDA/spack-stack#379]
Note: Please search to see if an issue already exists for the software you are requesting to be installed.

**Which software in the stack would you like installed?**
Please provide the name of the software you would like installed on the HPCs.  Ensure the software exists in the list of supported software.  If new software needs to be added to the stack, use the [add_package_request.md](./add_package_request.md) template.
FMS

**What is the version/tag of the software?**
Please provide the version/tag of the software to be installed.
FMS 2022.04 (https://github.com/NOAA-GFDL/FMS/releases/tag/2022.04)

**What compilation options would you like set?**
Give a detailed list of compilation options, compiler and MPI flavors and versions, etc.
Same options as used for FMS/2022.03
Intel and gnu

**Which machines would you like to have the software installed?**
See currently managed HPC machines.
All supported machines.
**Any other relevant information that we should know to correctly install the software??**
Please describe in as much detail as possible.

**Additional context**
Add any other context or screenshots about the install request here.

**Comment by mathomp4** [comment]
One request from the GMAO side. Could you add to the FMS spack `package.py` a `32BIT` variant:
```python
    variant(
        "32bit",
        default=False,
        description="Build a version of the library with default 32 bit reals",
    )
...
            self.define_from_variant("32BIT"),
```

Currently GEOS uses *both* `32BIT` and `64BIT` variants of FMS and I had to add that to the FMS I am using in my spack testing. (MOM6 is 64bit but our FV3 is built 32bit so we need two FMS's..fun fun!)

(NOTE: I know FMS has added that "mixed-mode" to the Autotools build, but we'll be doing CMake so we need both until I can figure out if there is a way for mixed-mode in CMake builds.)

**Comment by climbfuji** [comment]
Status 12/01/2022:
1. fms@2022.04 is already available through spack in our fork, see https://github.com/NOAA-EMC/spack/blob/a5127b4b834ffcb524dd7b34bed496a791bad962/var/spack/repos/builtin/packages/fms/package.py
2. fms@2022.04 is *not* the default for the JEDI applications (`skylab-dev` template). We rely on `fms@2022.02` for the upcoming `spack-stack@1.2.0` release on the JEDI side.
3. The outdated `hpc-stack-dev` template (meant to provide identical versions compared to `hpc-stack`) still has `fms@2022.01` as default, and so does https://github.com/NOAA-EMC/spack/blob/a5127b4b834ffcb524dd7b34bed496a791bad962/var/spack/repos/jcsda-emc-bundles/packages/ufs-weather-model-env/package.py.
4. The currently open UFS+Hera PR still has `fms@2022.01` in ufs-weather-model-env (see https://github.com/NOAA-EMC/spack/pull/192/files).

Unless @AlexanderRichert-NOAA wants to include the update to `fms@2022.04` in his PRs (with all the potential pitfalls of the multi-precision builds), this will not be included in `spack-stack@1.2.0` but made available in test versions towards the next release of spack stack in 2023.
